ros/angel_system_nodes/angel_system_nodes/question_answerer.py

```diff
@@ -0,0 +1,178 @@
+import json
+import openai
+import os
+import queue
+import rclpy
+from rclpy.node import Node
+import requests
+from termcolor import colored
+import threading
+from angel_msgs.msg import InterpretedAudioUserEmotion, SystemTextResponse
+
+
+openai.organization = os.getenv("OPENAI_ORG_ID")
+openai.api_key = os.getenv("OPENAI_API_KEY")
+
+IN_EMOTION_TOPIC = "user_emotion_topic"
+OUT_QA_TOPIC = "system_text_response_topic"
+FEW_SHOT_PROMPT = "few_shot_prompt_file"
+
+
+class QuestionAnswerer(Node):
+    def __init__(self):
+        super().__init__(self.__class__.__name__)
+        self.log = self.get_logger()
+
+        parameter_names = [IN_EMOTION_TOPIC, OUT_QA_TOPIC, FEW_SHOT_PROMPT]
+        set_parameters = self.declare_parameters(
+            namespace="",
+            parameters=[(p,) for p in parameter_names],
+        )
+        some_not_set = False
+        for p in set_parameters:
+            if p.type_ is rclpy.parameter.Parameter.Type.NOT_SET:
+                some_not_set = True
+                self.log.error(f"Parameter not set: {p.name}")
+        if some_not_set:
+            raise ValueError("Some parameters are not set.")
+
+        self._in_emotion_topic = self.get_parameter(IN_EMOTION_TOPIC).value
+        self._out_qa_topic = self.get_parameter(OUT_QA_TOPIC).value
+        self.prompt_file = self.get_parameter(FEW_SHOT_PROMPT).value
```

Comment:
What is the purpose of this prompt file?

Reply:
I felt it was a cleaner organization of the prompt text. All LLM prompts get parsed into the format listed in the specified PROMPT file.

Alternatively we could have a very long python string, which doesn't appear as nice and adds more lines to the .py file.